kiwi/markup/base.py

```diff
@@ -15,54 +15,39 @@
 # You should have received a copy of the GNU General Public License
 # along with kiwi.  If not, see <http://www.gnu.org/licenses/>
 #
-import os
 from tempfile import NamedTemporaryFile
 from lxml import etree
 
 # project
 from kiwi.defaults import Defaults
 
-from kiwi.exceptions import KiwiDescriptionInvalid
-
 
 class MarkupBase:
     """
     **Implements base class for Markup interface**
 
     Attributes
 
-    :param string description: description path or content
+    :param str description: description path or content
     """
-    def __init__(self, description):
-        description_is_file = False
-        if os.path.exists(description):
-            description_is_file = True
-        try:
-            if description_is_file:
-                self.description = description
-            else:
-                self.description_temporary = NamedTemporaryFile()
-                with open(self.description_temporary.name, 'wb') as outfile:
-                    outfile.write(description)
-                self.description = self.description_temporary.name
-        except Exception as issue:
-            raise KiwiDescriptionInvalid(issue)
+    def __init__(self, description: str):
+        self.description = description
```

Comment:
Makes sense this, I think it is a good idea to keep this description str variable as a single purpose. I see not much gain on keeping it just for extensions management. All in all the extensible XML feature is something to probably rethink, it turns to be quite complex extending the XML, I would even say far from obvious in a python virtual environment.